Subject: [PATCH] Fix media server crash in getFrameAtTime

Symptom: extract thumbnail cause media server crash
Root Cause: invoke media source start twice

diff --git a/media/libstagefright/StagefrightMetadataRetriever.cpp b/media/libstagefright/StagefrightMetadataRetriever.cpp
index ed7a2e6..f9b41fb 100644
--- a/media/libstagefright/StagefrightMetadataRetriever.cpp
+++ b/media/libstagefright/StagefrightMetadataRetriever.cpp
@@ -228,6 +228,7 @@
     if (err != OK) {
         ALOGW("source failed to start: %d (%s)", err, asString(err));
         decoder->release();
+        source->stop();
         return NULL;
     }
 
